src/cpp/src/rag/text_embedding_pipeline.cpp

```diff
@@ -0,0 +1,304 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "openvino/genai/rag/text_embedding_pipeline.hpp"
+
+#include "debug_utils.hpp"
+#include "openvino/genai/tokenizer.hpp"
+#include "openvino/op/broadcast.hpp"
+#include "openvino/op/concat.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/divide.hpp"
+#include "openvino/op/maximum.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/normalize_l2.hpp"
+#include "openvino/op/reduce_mean.hpp"
+#include "openvino/op/reduce_sum.hpp"
+#include "openvino/op/shape_of.hpp"
+#include "openvino/op/slice.hpp"
+#include "openvino/op/squeeze.hpp"
+#include "openvino/op/unsqueeze.hpp"
```

Comment:
You can use opset to get the operations, like here: https://github.com/openvinotoolkit/openvino_tokenizers/blob/979cba29ffc1f99d3c53e28a2c29b750390e59d9/src/tensorflow_translators.cpp#L9

Reply:
Opset headers used instead of exact ops.